src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,142 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ELEMENTDATA = {};
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList(int size) {
+        if (size > 0) {
+            this.elementData = new Object[size];
+            this.size = 0;
+        } else if (size == 0) {
+            this.elementData = EMPTY_ELEMENTDATA;
+            this.size = 0;
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity: " + size);
+        }
+    }
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= elementData.length) {
+            grow();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index " + index + "size " + size);
+        }
+        indexInRange(index, elementData.length - 1);
+        Object[] oldElemData = elementData;
+        Object[] newElemData = new Object[elementData.length + 1];
+        System.arraycopy(oldElemData, 0, newElemData, 0, index);
+        newElemData[index] = value;
+        System.arraycopy(oldElemData, index, newElemData, index + 1, size - index);
+        size++;
+        elementData = newElemData;
+        return;
```

Comment:
```suggestion
        System.arraycopy(elementData, index, elementData, index + 1, size - index);
        elementData[index] = value;
        size++;
```

Reply:
ok, added extra check before and it works, such way is much shorter and obviouse, thanks
![image](https://user-images.githubusercontent.com/85991807/163982895-9559e233-62de-4c48-9928-ab5a547a4306.png)